### DISCUSSION TITLE
PaoPao GateWay加paopaodns分流，配置好后无法连接外网

### DISCUSSION THREAD

**Discussion by wwushi** [kkkgo/PaoPaoGateWay#107] [问和答] [answered]
参考[https://github.com/kkkgo/PaoPaoGateWay/discussions/39](url)和[https://github.com/kkkgo/PaoPaoGateWay/discussions/23](url)配置好后无法连接外网，DNS也正常递归了，clash ui也能看到有连接，节点是通的，核心试过更换mihomo也是一样的情况，DNS和 GateWay采用的是vm虚拟机测试

**Comment by kkkgo** [comment, marked as answer]
首先用nslookup看看你目标域名是否被解析成fakeip，比如nslookup google.com
然后用traceroute跟踪一下路由，看看是否正常跑到fakeip网关去了。

**Comment by wwushi** [comment]
![Snipaste_2024-07-15_10-06-55](https://github.com/user-attachments/assets/6ad064c2-1d25-43f9-81ad-5865b8e3e433)
结果是这样，paopaoDNS ip192.168.90.253，GateWay ip192.168.90.254，主路由192.168.88.1/22

**Reply by kkkgo** [reply]
说明你的静态路由没做好。

**Comment by wwushi** [comment]
![Snipaste_2024-07-15_15-13-55](https://github.com/user-attachments/assets/53a75bef-db3c-4426-b3b9-a5ef792233b6)
![Snipaste_2024-07-15_15-14-13](https://github.com/user-attachments/assets/a192243b-3a3e-4922-ae67-c205152d9207)
静态路由我是这样设置的，跟参考教程一样也是TP主路由

**Reply by kkkgo** [reply]
你可以在windows上手动添加本地静态路由来测试：`route add 11.0.0.0 mask 255.0.0.0 192.168.90.254`

**Reply by wwushi** [reply]
手动指定测试还是不行，连主路由网关都追踪不到了。。。。。。

**Reply by kkkgo** [reply]
windows电脑是否和虚拟机在同一个交换机下，并且能互相ping通？
要详细分析的话，最好把你的ppgw.ini和虚拟机日志贴出来。

**Reply by wwushi** [reply]
今天莫名其妙可以了。。。。。我先用几天看看情况

**Reply by wwushi** [reply]
![Snipaste_2024-07-16_16-48-01](https://github.com/user-attachments/assets/b02340f6-78bd-402f-a8ed-5be0028e13d9)
想问下这个网关超时是指节点网关超时吗？

**Reply by kkkgo** [reply]
节点无响应

**Reply by wwushi** [reply]
频繁出现这个问题是什么原因呢？我直接ping网关，DNS，节点是通的

**Reply by kkkgo** [reply]
说明节点连接性差。fast_node在测试无法用节点访问test_node_url的时候会尝试重新测速选择节点，并从延迟低到高逐级筛选。以上测试均为真实连接网站测试。能否ping通和能否连接代理没有什么关系，ping是icmp协议，ping不通也可以连接，ping通也不能说明可以连接。

**Reply by wwushi** [reply]
但是我用v2rayn软件又能正常连接，不知道为啥，难受了

**Reply by kkkgo** [reply]
如果你始终不能连上就是协议不支持核心问题，如果能连上但频繁出现重新测速，说明节点连接不稳定，没有为啥，只是你用v2rayn软件不会每30秒测一次能不能用而已，你可以把fast_node功能关闭自己选节点。
